deps-mvn-indexing/src/main/java/com/connexta/osgeyes/index/JarPackagesIndexCreator.java

```diff
@@ -0,0 +1,129 @@
+
+/** Makes all packages contained in a jar indexed and searchable. */
+@Singleton
+@Named(JarPackagesIndexCreator.ID)
+public class JarPackagesIndexCreator implements IndexCreator {
+
+  // Useful to reference within package scope
+  static final String ID = "deps/jar-packages";
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(JarPackagesIndexCreator.class);
+
+  private static final IndexerField JAR_PACKAGES_FIELD =
+      new IndexerField(
+          MvnOntology.JAR_PACKAGES,
+          IndexerFieldVersion.V3,
+          MvnOntology.JAR_PACKAGES.getFieldName(),
+          MvnOntology.JAR_PACKAGES.getDescription(),
+          Field.Store.YES,
+          Index.ANALYZED);
+
+  private static final List<IndexerField> FIELDS = Collections.singletonList(JAR_PACKAGES_FIELD);
+
+  private static final List<String> DEPS = Collections.singletonList("min");
+
+  @Override
+  public String getId() {
+    return ID;
+  }
+
+  @Override
+  public List<String> getCreatorDependencies() {
+    return DEPS;
+  }
+
+  @Override
+  public Collection<IndexerField> getIndexerFields() {
+    return FIELDS;
+  }
+
+  @Override
+  public void populateArtifactInfo(ArtifactContext artifactContext) throws IOException {
+    final ArtifactInfo info = artifactContext.getArtifactInfo();
+    final Map<String, String> attributes = info.getAttributes();
+    final File artifactFile = artifactContext.getArtifact();
+    if (artifactFile != null && artifactFile.isFile() && artifactFile.getName().endsWith(".jar")) {
+      ZipHandle handle = null;
+      try {
+        handle = ZipFacade.getZipHandle(artifactFile);
+        List<String> paths =
+            handle.getEntries().stream()
+                .filter(s -> s.endsWith(".class"))
+                .map(
+                    s -> {
+                      String[] split = s.split("/");
+                      return String.join(".", Arrays.copyOf(split, split.length - 1));
+                    })
+                .distinct()
+                .collect(Collectors.toList());
+        attributes.put(JAR_PACKAGES_FIELD.getKey(), String.join(",", paths));
+      } finally {
+        try {
+          ZipFacade.close(handle);
+        } catch (Exception e) {
+          LOGGER.error("Could not close jar file properly", e);
+        }
+      }
+    }
+  }
+
+  @Override
+  public void updateDocument(ArtifactInfo artifactInfo, Document document) {
+    final MvnCoordinate currentPomCoord =
+        MvnCoordinate.newInstance(
+            artifactInfo.getGroupId(), artifactInfo.getArtifactId(), artifactInfo.getVersion());
+
+    final Map<String, String> attributes = artifactInfo.getAttributes();
+    final String pathText = attributes.get(JAR_PACKAGES_FIELD.getKey());
+
+    if (pathText == null) {
+      LOGGER.trace(
+          "No path text to write to lucene index for artifact {}",
+          MvnCoordinate.write(currentPomCoord));
+    } else {
+      document.add(JAR_PACKAGES_FIELD.toField(pathText));
```

Comment:
❓ Can't remember what our decision was regarding how to store the packages on the document. Since this field is fully tokenized is this the same as doing the following? Can we discover the packages individually either way?
```
final List<String> packages = Arrays.asList(pathText.split(","));
packages.forEach(p -> document.add(JAR_PACKAGES_FIELD.toField(p)));
```

Reply:
That could be done. It would produce a different document for each of the packages, eg.

```
mil.nga.gsr.catalog.aoi|aoi-gml-conversion|16.0.0-SNAPSHOT|NA|jar[bundle]
    mil.nga.gsr.catalog.aoi.gml.handler
mil.nga.gsr.catalog.aoi|aoi-kml-conversion|16.0.0-SNAPSHOT|NA|jar[bundle]
    mil.nga.gsr.catalog.aoi.kml.handler
```

Reply:
>It would produce a different document for each of the packages
```
mil.nga.gsr.catalog.aoi|aoi-gml-conversion|16.0.0-SNAPSHOT|NA|jar[bundle]
    mil.nga.gsr.catalog.aoi.gml.handler
mil.nga.gsr.catalog.aoi|aoi-kml-conversion|16.0.0-SNAPSHOT|NA|jar[bundle]
    mil.nga.gsr.catalog.aoi.kml.handler
```

How does that differ from how it operates now?

Reply:
With all packages in a comma-delisted list:
```
mil.nga.gsr.catalog.elasticsearch|elasticsearch-source|16.0.0-SNAPSHOT|NA|jar[bundle]
    ddf.catalog.impl.action,mil.nga.gsr.elasticsearch.oauth.credentialmanager,mil.nga.gsr.elasticsearch.source
```
Adding each package individually:
```
mil.nga.gsr.catalog.elasticsearch|elasticsearch-source|16.0.0-SNAPSHOT|NA|jar[bundle]
    ddf.catalog.impl.action
```

The individual approach seems to overwrite the value instead of making it a multi-value, so the other packages are lost.

Reply:
Sticking to the comma-separated list approach because otherwise values are lost.

Reply:
## Additional notes

### Staying with CSV will be fine for what we're trying to do:
- Field is tokenized so it will go through the default tokenization process.
- Currently used version of Lucene core (Tokenizer - https://lucene.apache.org/core/5_5_5/core/org/apache/lucene/analysis/Tokenizer.html) appears to have most of its behavior migrated correctly (StandardTokenizer - https://lucene.apache.org/core/6_6_0/core/org/apache/lucene/analysis/standard/StandardTokenizer.html).
- The 6.6.0 docs link to an external unicode spec; the rules about default word boundaries should work for our use case (http://unicode.org/reports/tr29/#Word_Boundaries).
- Behavior has been verified and proven out to be consistent with the above so we will keep this change; commas `,` act as a word boundary and dots `.` do not.

### Troubleshooting notes
- It seems adding the packages in multiple calls to `document.add` overwrites the prior ones because doing that only ever returns a single result.
- Changing the packages field to `NOT_ANALYZED` and indexing packages individually didn't fix it.
- If revisiting or retesting later, compare with how `modules` were done because those worked fine last they were tested.